parmed/structure.py

```diff
@@ -307,12 +307,12 @@ def __repr__(self):
                 retstr.append(' (triclinic)')
             else:
                 retstr.append(' (orthogonal)')
-        # Just assume that if the first bond has a defined type, so does
+        # Just assume that if the first atom has a defined type, so does
         # everything else... we don't want __repr__ to be super expensive
-        if len(self.bonds) > 0 and self.bonds[0].type is not None:
-            retstr.append('; parametrized>')
+        if self.atoms[0].type:
```

Comment:
```suggestion
        if self.atoms[0].atom_type is not UnassignedAtomType and self.atoms[0]._epsilon is None:
```

This is probably a better heuristic to use.  It's possible to provide parameters without defining an explicit type string, but you really can't provide parameters and have the atoms have an UnassignedAtomType *and* the nonbonded L-J terms be not set.